exporter/opentelemetry-exporter-otlp-proto-grpc/src/opentelemetry/exporter/otlp/proto/grpc/exporter.py

```diff
@@ -219,10 +220,15 @@ def __init__(
         parsed_url = urlparse(endpoint)
 
         if insecure is None:
-            if parsed_url.scheme == "https":
-                insecure = False
-            else:
+            insecure = environ.get(OTEL_EXPORTER_OTLP_INSECURE)
+
+        if parsed_url.scheme == "https":
+            insecure = False
+        if insecure is None:
+            if parsed_url.scheme == "http":
                 insecure = True
+            else:
+                insecure = False
 
```

Comment:
Take a case where scheme is not set and `OTEL_EXPORTER_OTLP_INSECURE` is set to False, then `insecure` would be string value instead of bool and `if insecure: ...` evaluates to true which is not expected?

Reply:
Thanks. There is no standard way to handle this in Python so I have adapted a pattern from `opentelemetry-python-contrib`:

```
            environ.get(OTEL_PYTHON_LOG_CORRELATION, "false").lower()
            == "true",
```

Elsewhere in that package there is this code:
```
        # FIXME this is probably a pattern that will show up in the rest of the
        # ext. Find a better way of implementing this.
        if environ.get(OTEL_PYTHON_DJANGO_INSTRUMENT) == "False":
            return
```
So it might be worth making a standardized way of handling boolean environment variables later.